agents/holo_synth/synth_agent.py

```diff
@@ -0,0 +1,234 @@
+        )
+
+        if config_file is None:
+            raise Exception("No config file specified for the FTS mirror config")
+        else:
+            config_file_path = os.path.join(os.environ["OCS_CONFIG_DIR"], config_file)
+
+            with open(config_file_path) as stream:
+                self.holog_configs = yaml.safe_load(stream)
+                if self.holog_configs is None:
+                    raise Exception("No mirror configs in config file.")
+                self.log.info(f"Loaded mirror configs from file {config_file_path}")
+                self.N_MULT = self.holog_configs.pop("N_MULT", None)
+                self.ghz_to_mhz = self.holog_configs.pop("ghz_to_mhz", None)
+
+    def init_synth(self, session, params=None):
+        """init_synth(params=None)
+
+        **Task** - A task to initialize the synthesizers.
+
+        Parameters: None
+
+        Examples:
+            Example for calling in a client::
+                import ocs
+                from ocs.ocs_client import OCSClient
+                agent = OCSClient("synth_LO")
+                agent.init_synth()
+
+        Notes:
+            This task is called to turn on the synthesizers.
+        """
+
+        self.log.debug("Trying to acquire lock")
+        with self.lock.acquire_timeout(timeout=0, job="init") as acquired:
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn(
+                    "Could not start init because {} is already running".format(
+                        self.lock.job
+                    )
+                )
+                return False, "Could not acquire lock."
+            # Run the function you want to run
+            self.log.debug("Lock Acquired Connecting to Stages")
+            self.lo_id = synth3.get_LOs()
+
+            synth3.set_RF_output(0, 1, self.lo_id)  # LO ID, On=1, USB connection ID
+            synth3.set_RF_output(1, 1, self.lo_id)  # LO ID, On=1, USB connection ID
+
+            # data = {"timestamp": time.time(), "block_name": "synth_LO", "data": {}}
+
+            # data["data"]["F1_status"] = 1
+            # data["data"]["F2_status"] = 1
+
+            # self.agent.publish_to_feed("synth_LO", data)
+            # session.data.update(data["data"])
+
+        # This part is for the record and to allow future calls to proceed,
+        # so does not require the lock
+        self.initialized = True
+
+        return True, "Synth Initialized."
+
+    @ocs_agent.param("offset", type=float, default=0, check=lambda x: 0 <= x <= 1000)
+    @ocs_agent.param("freq1", type=float, default=0, check=lambda x: 0 <= x <= 1000)
+    def set_frequencies(self, session, params):
+        """set_frequencies(freq1=0, offset=0)
+
+        **Task** - A task to set the frequencies of the synthesizers.
+
+        Parameters:
+            dict: {'freq1': float
+                   'offset': float}
```

Comment:
These parameters should be documented like typical keyword arguments, not the dict that `params` really is. So this should be:
```
Parameters:
    freq1 (float): Description of what freq1 is, note its units here.
    offset (float): Description of what offset is, also noting its units.
```

You can then drop the `Notes:` section, as the units will be documented in the individual arguments, and the user doesn't really need to know the details of the conversion.